src/librustc_codegen_ssa/back/link.rs

```diff
@@ -765,20 +765,21 @@ fn link_sanitizer_runtime(sess: &Session, crate_type: config::CrateType, linker:
     let default_sysroot = filesearch::get_or_default_sysroot();
     let default_tlib =
         filesearch::make_target_lib_path(&default_sysroot, sess.opts.target_triple.triple());
+    let channel = option_env!("CFG_RELEASE_CHANNEL").unwrap_or("unknown");
```

Comment:
Please create librustc_codegen_ssa/build.rs with the following contents:

```
    println!("cargo:rerun-if-changed=build.rs");
    println!("cargo:rerun-if-env-changed=CFG_RELEASE_CHANNEL");
```

I also am not sure about the "unknown" default here, maybe it makes sense to instead omit the channel entirely? In practice we should always have the channel configured during published builds so it shouldn't matter.

Reply:
Sure, channel is now ignored if empty, even if that should never happen